internal/rbd/rbd_util.go

```diff
@@ -1924,10 +1926,12 @@ func genVolFromVolIDWithMigration(
 
 // setVolumeMetadata set PV/PVC/PVCNamespace metadata on RBD image.
 func (rv *rbdVolume) setVolumeMetadata(parameters map[string]string) error {
-	for k, v := range k8s.GetVolumeMetadata(parameters) {
-		err := rv.SetMetadata(k, v)
-		if err != nil {
-			return fmt.Errorf("failed to set metadata key %q, value %q on image: %w", k, v, err)
+	if rv.EnableMetadata {
```

Comment:
Do the reverse check, and return if setting the metadata is not needed. It is like testing for an error, and returning before executing further operations.

That reduces the indention for the next code block. It is the style we try to use throughout the project.